includes/documents/class-wcpdf-invoice.php

```diff
@@ -17,16 +19,31 @@
  */
 
 class Invoice extends Order_Document_Methods {
+	
+	public $type;
+	public $title;
+	public $icon;
+	public $lock_name;
+	public $lock_context;
+	public $lock_time;
+	public $lock_retries;
+	
 	/**
 	 * Init/load the order object.
 	 *
 	 * @param  int|object|WC_Order $order Order to init.
 	 */
 	public function __construct( $order = 0 ) {
 		// set properties
-		$this->type		= 'invoice';
-		$this->title	= __( 'Invoice', 'woocommerce-pdf-invoices-packing-slips' );
-		$this->icon		= WPO_WCPDF()->plugin_url() . "/assets/images/invoice.svg";
+		$this->type         = 'invoice';
+		$this->title        = __( 'Invoice', 'woocommerce-pdf-invoices-packing-slips' );
+		$this->icon         = WPO_WCPDF()->plugin_url() . "/assets/images/invoice.svg";
+		
+		// semaphore
+		$this->lock_name    = "wpo_wcpdf_{$this->slug}_semaphore_lock";
+		$this->lock_context = array( 'source' => "wpo-wcpdf-{$this->type}-semaphore" );
+		$this->lock_time    = apply_filters( 'wpo_wcpdf_semaphore_lock_time', 2 );
+		$this->lock_retries = apply_filters( 'wpo_wcpdf_semaphore_lock_retries', 0 );
```

Comment:
Perhaps we need to add the document type to these filter names? So if necessary we can change lock time and lock retries independently from the lock time and retries of the yearly invoice number reset.

So something like: 
```
wpo_wcpdf_{$this->slug}_number_lock_time
wpo_wcpdf_{$this->slug}_number_lock_retries
```